validator/src/commands/run/args.rs

```diff
@@ -442,11 +443,13 @@ pub fn add_args<'a>(app: App<'a, 'a>, default_args: &'a DefaultArgs) -> App<'a,
             .value_name("NUMBER")
             .takes_value(true)
             .default_value(&default_args.full_snapshot_archive_interval_slots)
+            .validator(is_non_zero)
```

Comment:
I think newer CLAP might have some built-ins for this ... some day ha

Regardless, I think we might get this validation for free if we change this to `NonZeroU64`
https://github.com/anza-xyz/agave/blob/40c7522a754eb44fb0b43450e4a249086edab1e2/validator/src/commands/run/execute.rs#L934

Arguably, we should plumb `NonZero` all the way through, but that could be separate and we could use `NonZeroU64::get()` right here to keep the scope of this PR focused